docs/data/data-grid/column-menu/column-menu.md

```diff
@@ -0,0 +1,130 @@
+
+It can be accessed by clicking on 3-dots icon that appear on hover on a header cell.
+
+{{"demo": "ColumnMenuGrid.js", "bg": "inline"}}
+
+## Disabled column menu
+
+By default, each column header displays a column menu. The column menu allows actions to be performed in the context of the target column, e.g. filtering. To disable the column menu, set the prop `disableColumnMenu={true}`.
+
+{{"demo": "DisabledColumnMenuGrid.js", "bg": "inline"}}
+
+## Simple column menu
+
+In case you like the look and feel of the old column menu, you can replace the default one with `GridColumnMenuSimple` component.
+
+{{"demo": "SimpleColumnMenuGrid.js", "bg": "inline"}}
+
+## Customise column menu items
+
+You can also customise column menu based on some conditions. Every item in the menu is assigned a `slot` using which it's possible to:
+
+- Hide/Show specific items
+- Override default items (if you want to override one item and not the whole menu)
+- Add new items
+- Change display order for items
+
+For that purpose you can use `getVisibleColumnMenuItems` and `columnMenuItems`.
+
+**getVisibleColumnMenuItems**: It is available in both `GridColDef` and `componentsProps.columnMenu`, it receives list of all registered slots and should return a `filtered` and `ordered` list of items that are needed to be shown. It can be used either per-column basis by passing in `GridColDef` or for whole Grid by passing in `componentsProps.columnMenu`. If you have it in both, the preference will be given to the `GridColDef` one.
+
+**columnMenuItems**: It could be used to override or register new items to the menu. You can simply pass the object with existing or new items and they will be updated/added to the grid.
+
+### Hide/Show Specific Items:
+
+For every column menu component there's a default order of items configured by default called `visibleMenuItems`, for default column menu it's:
+
+```tsx
+[
+  'sorting',
+  'divider',
+  'filter',
+  'divider',
+  'hideColumn',
+  'divider',
+  'manageColumns',
+];
+```
+
+Using `getVisibleColumnMenuItems` method, you can override this order based on configured slots for the column menu.
+
+```tsx
+const getVisibleColumnMenuItems = () => ['sorting', 'divider', 'filter']; // only show `sort` & `filter`
+
+return (
+  <DataGrid
+    {...data}
+    componentsProps={{ columnMenu: { getVisibleColumnMenuItems } }}
+  />
+);
+```
+
+### Override default items and add new items:
+
+Using `componentsProps.columnMenu.columnMenuItems`, you can:
+
+1. Override Existing Items
+2. Add new Items
+
+```tsx
+const columnMenuItems = {
+  ['filter']: {
+    // overiding existing slot
+    component: <MyCustomFilter />, // overriden property
+  },
+  ['closeMenu']: {
+    // adding new slot
+    component: <MenuCloseComponent />,
+    displayName: 'MenuClose',
+  },
+};
+
+// add new item in visible items and append it to the last of list
+const getVisibleColumnMenuItems = (defaultItems) => [...defaultItems, 'closeMenu'];
```

Comment:
There's if the user doesn't use `getVisibleColumnMenuItems`, there's a default order configured for each Column Menu component, but in case user wants to *filter*, *reorder* or *add new* they can use `getVisibleColumnMenuItems` in which users will have to specify complete order.
In case they just want to append to the end of menu, they can just use `defaultItems` passed in function and concatenate the new items like above.

Reply:
Maybe rephrase this part then?

```diff
- it receives keys of all registered items
+ it receives keys of default menu items
```

Reply:
Thanks for the feedback, I'm trying to update its interface and in args these values will be passed: `currentColumn`, `visibleItemKeys` (items currently visible) and `itemKeys` (all the available keys) so that if user wants to do some conditional rendering based on colDef or visible or all available items.